apps/web/src/components/ParameterInput/index.tsx

```diff
@@ -21,31 +21,93 @@ export function ParameterInput({
   disabled = false,
 }: {
   type: ParameterType
-  value: string
+  value?: string
   onChange: (value: string) => void
   disabled?: boolean
 }) {
-  const { uploadFile, isLoading } = useFiles()
+  if (type === ParameterType.File || type === ParameterType.Image) {
+    return (
+      <FileParameterInput
+        type={type}
+        value={value}
+        onChange={onChange}
+        disabled={disabled}
+      />
+    )
+  }
+
+  if (type === ParameterType.Text) {
+    return (
+      <TextParameterInput
+        value={value}
+        onChange={onChange}
+        disabled={disabled}
+      />
+    )
+  }
+
+  return (
+    <TextArea
+      value='Parameter type not supported'
+      minRows={1}
+      maxRows={1}
+      disabled={true}
+    />
+  )
+}
 
+function TextParameterInput({
+  value,
+  onChange,
+  disabled = false,
+}: {
+  value?: string
+  onChange: (value: string) => void
+  disabled?: boolean
+}) {
   const onTextChange = useCallback(
     (e: ChangeEvent<HTMLTextAreaElement>) => {
       onChange(e.target.value)
     },
     [onChange],
   )
 
-  const onImageChange = useCallback(
-    async (files: FileList | null) => {
-      const file = files?.[0]
-      if (file) {
-        const uploadedFile = await uploadFile({ file })
-        if (uploadedFile) return onChange(JSON.stringify(uploadedFile))
+  const textValue = useMemo(() => {
+    try {
+      const file = JSON.parse(value || '')
+      if (file?.url) {
+        return file.url
       }
+    } catch {
+      // Do nothing
+    }
 
-      onChange('')
-    },
-    [uploadFile, onChange],
+    return value
+  }, [value])
+
+  return (
+    <TextArea
+      defaultValue={textValue}
```

Comment:
this is the only change in this file (the rest is just splitting components), changed value to defaultValue